Packs/CybleThreatIntel/Integrations/CybleThreatIntel/CybleThreatIntel.py

```diff
@@ -4,151 +4,314 @@
+        self.client = create_client(
+            self.parsed_url.netloc,
+            use_https=True,
+            discovery_path=self.parsed_url.path)
+        self.client.set_auth(username=self.username, password=self.password)
+
+    def __fetch(self, begin, end, collection):
+        for block in self.client.poll(collection_name=collection, begin_date=begin, end_date=end):
+            yield block.content.decode('utf-8')
+
+    def get_recursively(self, search_dict, field):
+        """
+        Takes a dict with nested lists and dicts,
+        and searches all dicts for a key of the field
+        provided.
+        """
+        fields_found = []
+        for key, value in iter(search_dict.items()):
+
+            if key == field:
+                fields_found.append(value)
+
+            elif isinstance(value, dict):
+                results = self.get_recursively(value, field)
+                for result in results:
+                    fields_found.append(result)
+
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        more_results = self.get_recursively(item, field)
+                        for another_result in more_results:
+                            fields_found.append(another_result)
+
+        return fields_found
+
+
+    def build_indicators(self, args, data):
+        indicators= []
+        for eachres in data:
+            indicator_obj = {
+                "service": "Cyble Feed"
+            }
+            for eachtype in FeedIndicatorType.list_all_supported_indicators():
+                if eachtype.lower() in args.get('collection').lower():
+                    indicator_obj['type'] = eachtype
+                    break
+
+            data = self.get_recursively(eachres, 'value')
+
+            ind_val = {}
+            for eachindicator in data:
+                typeval = auto_detect_indicator_type(eachindicator)
+                if typeval:
+                    ind_val[typeval] = eachindicator
+
+            if len(data) == 1:
+                indicator_obj['value'] = str(data[0])
+            elif indicator_obj['type'] in list(ind_val.keys()):
+                indicator_obj['value'] = str(ind_val[indicator_obj['type']])
+            elif len(ind_val) != 0:
+                indicator_obj['type'] = list(ind_val.keys())[0]
+                indicator_obj['value'] = ind_val[list(ind_val.keys())[0]]
+
+            if eachres.get('indicators'):
+                for eachindicator in eachres.get('indicators'):
+                    indicator_obj['title'] = eachindicator.get('title')
+                    indicator_obj['time'] = eachindicator.get('timestamp')
+            else:
+                for eachindicator in eachres.get('ttps').get('ttps'):
+                    indicator_obj['title'] = eachindicator.get('title')
+                    indicator_obj['time'] = eachindicator.get('timestamp')
```

Comment:
In both options you lose all values except the last ones,
Are you sure this should be the case?